pkg/proc/native/proc.go

```diff
@@ -270,6 +270,7 @@ func (dbp *nativeProcess) handlePtraceFuncs() {
 	// while invoking the ptrace(2) syscall. This is due to the fact that ptrace(2) expects
 	// all commands after PTRACE_ATTACH to come from the same thread.
 	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
```

Comment:
I think I like that the thread is not reused after we're done with it, I would prefer this to be:

```
if runtime.GOOS == "openbsd" { 
     // workaround for https://github.com/golang/go/issues/52394
    defer runtime.UnlockOSThread()
}
```

Reply:
Sure, done. Alternatively, as mentioned above we could also avoid calling `runtime.LockOSThread` in the first place on these platforms.